### PR TITLE
Added multi parent realms. Added context resources and transaction.

### PR DESCRIPTION
**Pull request by bFekete**
Adds the ability to create a realm that is nested within multiple realms.

Example of realm hierarchy:

- LockOutRealm
  - CombinedRealm
    - JNDIRealm1
    - JNDIRealm2
    - JNDIRealm3

To specify multiple parents, use the parent realm parameter and specify them in order separated by comma. I did this to ensure I wouldn't break any existing implementation of this module.
```
 parent_realm => 'org.apache.catalina.realm.LockOutRealm,org.apache.catalina.realm.CombinedRealm'
```

Adds the \<Resources\> element within context.xml

Adds different type of \<Resource\> elements within context.xml. Ex: \<PreResources\>, \<PostResources\>

Adds the \<Transaction\> element within context.xml

### CODE REVIEW

**Comment by david22swan** [comment]
@bFekete Apologies for the bother but I'm just checking up on whether this is an ongoing piece of work or if you are ready to be reviewed?

**Comment by bFekete** [comment]
@david22swan ready for review.

File: manifests/config/server/realm.pp [hunk 1]
```diff
@@ -34,7 +34,7 @@
   $parent_service                                         = 'Catalina',
   $parent_engine                                          = 'Catalina',
   $parent_host                                            = undef,
-  $parent_realm                                           = undef,
+  $parent_realm                                           = undef, # if more than 2 levels of nested realms, specify and separate by comma
```
Comment (david22swan) [inline manifests/config/server/realm.pp:37]: Please add this note to the variable description at the top.

File: manifests/config/context/resources.pp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+# @param additional_attributes
+#   Specifies any further attributes to add to the Host. Valid options: a hash of '< attribute >' => '< value >' pairs.
+# @param attributes_to_remove
+#   Specifies an array of attributes to remove from the element. Valid options: an array of strings.
+# @param show_diff
+#   Specifies display differences when augeas changes files, defaulting to true. Valid options: true or false.
+#
+define tomcat::config::context::resources(
+  Optional[String] $catalina_base     = undef,
+  Optional[String] $resources_name    = undef,
```
Comment (david22swan) [inline manifests/config/context/resources.pp:16]: This type need's to be added above

File: manifests/config/context/transaction.pp [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# @summary Configure Transaction elements in $CATALINA_BASE/conf/context.xml
+#
+# @param catalina_base
+#   Specifies the base directory of the Tomcat installation to manage. Valid options: a string containing an absolute path.
+# @param transaction_name
+#   The name of the transaciton.
```
Comment (david22swan) [inline manifests/config/context/transaction.pp:6]: Should state that it defaults to the $name

File: manifests/config/server/executor.pp [hunk 4]
```diff
@@ -0,0 +1,73 @@
+# @summary Configure Executor elements in $CATALINA_BASE/conf/server.xml
+#
+# @param catalina_base
+#   Specifies the base directory of the Tomcat installation to manage. Valid options: a string containing an absolute path.
+# @param executor_name
+#   The name of the executor element.
```
Comment (david22swan) [inline manifests/config/server/executor.pp:6]: As above, should state that it defaults to the name

**Comment by david22swan** [comment]
@bFekete Aside from a few small comments about your documentation this looks good to me. Would just like another team member to sign of on it as well.

**Comment by david22swan** [comment]
@bFekete Have you had a chance to look at the doc changes I have suggested?

**Comment by daianamezdrea** [comment]
Hi @bFekete, @david22swan, because this PR didn't have any activity for the last few months from the contributor, I'll close it, thank you! Feel free to open it again anytime, or add a new PR. Cheers !

**Pull request closed without merging**